src/gui/color_dialog.cpp

```diff
@@ -839,5 +871,23 @@ void ColorDialog::rgbValueChanged()
 	}
 }
 
+void ColorDialog::colorUsageItemClicked(QListWidgetItem* clicked_item)
+{
+	auto item_data = clicked_item->data(Qt::UserRole);
+	Q_ASSERT(item_data.canConvert<const Symbol*>());
+	auto* symbol = const_cast<Symbol*>(item_data.value<const Symbol*>());
+
+	auto* map = const_cast<Map*>(&this->map);
```

Comment:
These two instances of `const_cast` are a hint: Originally, `ColorDialog` is designed to not change the map. The click-through feature now enables direct changes (to the symbols). IMO we either have to change the relationship to the map, or we don't add this click-through feature at the moment. I tend to prefer the latter.

(And changing the symbols implies the need to update the color usage list.)

Reply:
Thanks for the review, Kai, I acknowledge the approach and I've updated the pull request.